runbotics/runbotics-orchestrator-ui/src/src-app/views/process/ProcessBuildView/Modeler/ActionFormPanel/widgets/SuggestProcessesNameWidget.tsx

```diff
@@ -1,54 +1,68 @@
-import React, { FC, useMemo, useEffect } from 'react';
+import React, { FC, useMemo, useEffect, useRef, useState } from 'react';
 
 import { TextField, Autocomplete } from '@mui/material';
 import { WidgetProps } from '@rjsf/core';
 
-import { IFormData } from '#src-app/Actions/types';
+import { BotSystem } from 'runbotics-common';
+
+import useTranslations from '#src-app/hooks/useTranslations';
 import { useDispatch, useSelector } from '#src-app/store';
 import { processActions } from '#src-app/store/slices/Process';
 
-interface CustomWidgetProps extends WidgetProps {
-    formData: IFormData;
-}
-
-const ProcessNameSuggestionWidget: FC<CustomWidgetProps> = (props) => {
+const ProcessNameSuggestionWidget: FC<WidgetProps> = (props) => {
     const dispatch = useDispatch();
     const { byId: processes } = useSelector((state) => state.process.all);
+    const [customError, setCustomError] = useState('');
+    const [value, setValue] = useState(props.value);
+    const { translate } = useTranslations();
 
     useEffect(() => {
         dispatch(processActions.getProcesses());
     // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
     const {
-        process: { id: processId },
+        process: { id: processId, system: { name: parentProcessSystem } },
     } = useSelector((state) => state.process.draft);
 
+    const isProcessSystemCompatible = (processSystem: string): boolean => (
+        parentProcessSystem === processSystem ||
+        parentProcessSystem === BotSystem.ANY ||
+        processSystem === BotSystem.ANY
+    );
+
     const options = useMemo(
         () =>
             processes
                 ? Object.values(processes)
-                    .filter((process) => process.id !== processId)
-                    .map<Number>((process) => process.id)
+                    .filter((process) => (process.id !== processId && isProcessSystemCompatible(process.system.name)))
+                    .map<number>((process) => process.id)
```

Comment:
to iterate once instead of two (filter + map = O(2n)) use reduce:

.reduce(acc, process) => <yourFilterConditions> ? return [...acc, process.id] : return acc